### ISSUE TITLE
Implementing Auto Login with SSO

### ISSUE THREAD

**Issue by WeeHong** [Adldap2/Adldap2-Laravel#701]
- Laravel Version: 5.8
- Adldap2-Laravel Version: 6.0
- PHP Version: 7.1
- LDAP Type: ActiveDirectory

### Description:
Hi,

I am currently developing an API which allows our company staff to login the website without entering their user credentials. The user credential will be fetch on the background when user open the web application.

I found that actually SSO can do the trick but I quite lost about how to implement it into the Laravel.

I check my Windows Authentication has been enabled but I don't know what to do next.

Do you have any tutorial which can help me to understand more how to implement it?

**Comment by stevebauman** [comment]
Hi @WeeHong, I'd be glad to help you out, though I'll need to know a couple things first.

Are you using IIS as your web-server? If so, there's a walk through guide here to enable it on your IIS instance:

https://docs.microsoft.com/en-us/iis/configuration/system.webServer/security/authentication/windowsAuthentication/providers/add

As well as some documentation I created here:

https://adldap2.github.io/Adldap2-Laravel/#/auth/middleware

Basically, once you enable Windows Authentication on your IIS application instance and you visit that site while logged in on a domain joined computer that your server is also on, the users domain account will be authenticated against it, and if successful, PHP will set a server variable equal to the domain users username (`$_SERVER['AUTH_USER']).

**Comment by WeeHong** [comment]
@stevebauman 
Thank you for providing such informative detail.

I would like to try it out first and get back to you as soon as possible.
I am also planning to write an article on Medium if possible.

Thank you.

**Comment by stevebauman** [comment]
No problem @WeeHong! Let me know if you need any more help. I'd love to read your article once your done too!

I'll re-open this issue if you need anything else.

**Comment by WeeHong** [comment]
Hi @stevebauman 

Sorry I have a dumb question about this.

If I am using Ngnix or Apache2 to host my website, I should use another solution right?

Thanks.

**Comment by stevebauman** [comment]
Hi @WeeHong,

Unfortunately I'm not sure, I've never set up SSO on Nginx or Apache, only IIS on Windows Server. Sorry!
